mars_troughs/custom_acc_models.py

```diff
@@ -83,11 +87,20 @@ def get_xt(
             horizontal distances (np.ndarray) of the same size as time input, in
             meters.
         """
+        
         yt = self.get_yt(time)
 
-        return -cot_angle * yt + csc_angle * (
-            int_retreat_model_t_spline(time) - int_retreat_model_t_spline(0)
-        )
+        if "Lag" in str(retrModel):
+                
+            return -cot_angle * yt + csc_angle * (
+                retr(time) - retr(0)
+            )
+        
+        elif "Retreat" in str(retrModel):  
+            out = cot_angle * (-yt)  + csc_angle * (retr)
+    
+            return out
+
```

Comment:
For readibility's sake I recommend writing it this way:
```python
left_product = -cot_angle * yt

if "Lag" in str(retrModel):
    return left_product  + csc_angle * (retr(time) - retr(0))
else:
    return left_product + csc_angle * retr
```

This removes some duplication.